tests/unit/test_base.py

```diff
@@ -45,14 +47,61 @@ def test_test_invalid():
     with pytest.raises(tmt.utils.GeneralError):
         test = tmt.Test({})
     # Invalid name
-    with pytest.raises(tmt.utils.SpecificationError):
+    with pytest.raises(SpecificationError):
         test = tmt.Test({}, name='bad')
     # Invalid attributes
     for key in ['component', 'require', 'tag']:
-        with pytest.raises(tmt.utils.SpecificationError):
+        with pytest.raises(SpecificationError):
             test = tmt.Test({key: 1}, name='/smoke')
-    with pytest.raises(tmt.utils.SpecificationError):
+    with pytest.raises(SpecificationError):
         test = tmt.Test({'environment': 'string'}, name='/smoke')
     # Listify attributes
     assert tmt.Test({'tag': 'a'}, name='/smoke').tag == ['a']
     assert tmt.Test({'tag': ['a', 'b']}, name='/smoke').tag == ['a', 'b']
+
+def test_link():
+    """ Test the link attribute parsing """
+    # No link should default to an empty list
+    assert Link().get() == []
+
+    # Single string (default relation)
+    assert Link('/fmf/id').get() == [dict(relates='/fmf/id')]
+    # Multiple strings (default relation)
+    assert Link(['one', 'two']).get() == [
+            dict(relates='one'), dict(relates='two')]
+    # Multiple string mixed relation
+    assert Link(['implicit', dict(duplicates='explicit')]).get() == [
+        dict(relates='implicit'), dict(duplicates='explicit')]
+    # Multiple strings (explicit relation)
+    family = [dict(parent='mon'), dict(child='son')]
+    assert Link(family).get() == family
+
+    # Single dictionary (default relation)
+    assert Link(dict(name='foo')).get() == [dict(relates=dict(name='foo'))]
+    # Single dictionary (explicit relation)
+    assert Link(dict(verifies='foo')).get() == [dict(verifies='foo')]
+    # Multiple dictionaries
+    family = [dict(parent='mom', note='foo'), dict(child='son')]
+    assert Link(family).get() == family
+
+    # Selected relations
+    assert Link(family).get('parent') == [dict(parent='mom', note='foo')]
+    assert Link(family).get('child') == [dict(child='son')]
+
+    # Full fmf id
+    fmf_id = tmt.utils.yaml_to_dict("""
+        blocked-by:
+            url: https://github.com/psss/fmf
+            name: /stories/select/filter/regexp
+        note: Need to get the regexp filter working first.
+        """)
+    link = Link(fmf_id)
+    assert link.get() == [fmf_id]
+
+    # Invalid links and relations
```

Comment:
I believe this should raise SpecificationError. Content of main.fmf:
```
test: echo
link:
  url: URL
  name: foo
  verifies: bar
```
Current output follow. 'url' is not a valid relation.
 ```
/
        test echo
        path /
      manual no
    duration 5m
     enabled yes
      result respect
         url URL
```

Reply:
Good catch!

Reply:
Should now be fixed by bc51d6e.